src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -247,6 +245,14 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
                     kind.MethodKind.HasValue ? SymbolKind.Method :
                     throw ExceptionUtilities.Unreachable;
 
+                // Suggest names from existing overloads.
+                if (symbolKind == SymbolKind.Parameter)
+                {
+                    var partialSemanticModel = await document.GetPartialSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                    if (partialSemanticModel is not null)
+                        AddNamesFromExistingOverloads(context, partialSemanticModel, result, cancellationToken);
```

Comment:
Caller already has a semantic model, but it's not a partial semantic model.

```csharp
var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
```

I used partial semantic model per your suggestion @sharwell.